Kitodo-API/src/main/java/org/kitodo/config/Config.java

```diff
@@ -11,141 +11,197 @@
+     *
+     * @return the directory for the process directories
      */
     public static String getKitodoDataDirectory() {
-        return getParameter(METADATA_DIRECTORY);
+        return getParameter(DIR_PROCESSES);
     }
 
     /**
-     * Get Kitodo config directory.
+     * Returns the directory that contains XML configuration files.
      *
-     * @return String
+     * @return the directory for XML configuration files
      */
     public static String getKitodoConfigDirectory() {
-        return getParameter(CONFIG_DIR);
+        return getParameter(DIR_XML_CONFIG);
     }
 
     /**
-     * Request selected parameter from configuration.
+     * Returns the selected parameter from the configuration file. Throws a
+     * {@code NoSuchElementException} if no such parameter exists.
      *
-     * @return Parameter as String
+     * @param key
+     *            key whose value is to be returned
+     * @return value for the requested key
+     * @throws NoSuchElementException
+     *             if parameter taken from config file is null or exception
+     *             occurred
      */
-    public static String getParameter(String parameter) {
+    public static String getParameter(String key) {
         try {
-            return getConfig().getString(parameter);
+            return getConfig().getString(key);
         } catch (NoSuchElementException e) {
-            throw new NoSuchElementException(
-                    "No configuration found in kitodo_config.properties for key " + parameter + "!");
+            logger.catching(e);
+            throw new NoSuchElementException("No configuration found in kitodo_config.properties for key " + key + "!");
         }
     }
 
     /**
-     * Request selected parameter with given default value from configuration.
+     * Returns the selected parameter from the configuration file. If no such
+     * parameter exists, returns the given default value.
      *
-     * @param parameter
-     *            name of parameter in config file
-     * @param defaultIfNull
-     *            default value in case parameter taken from config file is null or
-     *            exception occurred
-     * @return Parameter as String
+     * @param key
+     *            key whose value is to be returned
+     * @param defaultValue
+     *            default value in case parameter taken from config file does
+     *            not exist or exception occurred
+     * @return value for the requested key, or {@code defaultIfNull} if not
+     *         found
      */
-    public static String getParameter(String parameter, String defaultIfNull) {
+    public static String getParameter(String key, String defaultValue) {
         try {
-            return getConfig().getString(parameter, defaultIfNull);
+            return getConfig().getString(key, defaultValue);
         } catch (ConversionException e) {
-            return defaultIfNull;
+            logger.catching(e);
```

Comment:
Two different loggers in one catch block?

Reply:
This was there already. I only added logging of caught exception. Because as to coding guidelines, exception being caught must be logged.

Reply:
Is it necessary to double log it? I think one logger on level error is enough.

Reply:
This is different log levels, so in productive envirunment only the warning will be logged, but not a stack trace.